src/utils/promptRunner.ts

```diff
@@ -0,0 +1,107 @@
+    }
+
+    const baseArgs: string[] = [
+        'run',
+        '--rm',
+        '-v', '/var/run/docker.sock:/var/run/docker.sock',
+        '-v', 'openai_key:/root',
+        '--mount', 'type=volume,source=docker-prompts,target=/prompts'
+    ];
+
+    const runArgs: string[] = render ? [] : [
+        'vonwig/prompts:latest',
+        ...(render ? [] : ['run']),
+        "--host-dir", projectDir,
+        "--user", username,
+        "--platform", platform,
+        ...promptArgs,
+        '--jsonrpc',
+        '--pat', pat
+    ];
+    return [...baseArgs, ...mountArgs, ...runArgs];
+};
+
+const runAndStream = async (command: string, args: string[], callback: (json: any) => void) => {
+    output.appendLine(`Running ${command} with args ${args.join(' ')}`);
+    const child = spawn(command, args);
+
+    const onChildSTDIO = ({ stdout, stderr }: { stdout: string; stderr: string | null }) => {
+        if (stdout && stdout.startsWith('Content-Length:')) {
+
+            const rpcMessages = stdout.split('Content-Length: ').filter(Boolean).map(rpcMessage => rpcMessage.trim().slice(rpcMessage.indexOf('{')));
+
+            for (const rpcMessage of rpcMessages) {
+                let json;
+                try {
+                    json = JSON.parse(rpcMessage);
+                } catch (e) {
+                    console.error(`Failed to parse JSON: ${rpcMessage}, ${e}`);
+                    child.kill();
+                }
+                callback(json);
+
+
+            }
+        }
+        else if (stderr) {
+            callback({ method: 'error', params: { content: stderr } });
+        }
+        else {
+            callback({ method: 'message', params: { content: stdout } });
+        }
+    };
+    return new Promise((resolve, reject) => {
+        child.stdout.on('data', (data) => {
+            onChildSTDIO({ stdout: data.toString(), stderr: '' });
+        });
+        child.stderr.on('data', (data) => {
+            onChildSTDIO({ stderr: data.toString(), stdout: '' });
+        });
+        child.on('close', (code) => {
+            callback({ method: 'message', params: { debug: `child process exited with code ${code}` } });
+            resolve(code);
+        });
+        child.on('error', (err) => {
+            callback({ method: 'error', params: { content: JSON.stringify(err) } });
+            reject(err);
+        });
+    });
+};
+
+export const spawnPromptImage = async (promptArg: string, projectDir: string, username: string, platform: string, pat: string, callback: (json: any) => void) => {
+    const args = getRunArgs(promptArg!, projectDir!, username, platform, pat);
+    return runAndStream("docker", args, callback);
+};
+
+export const writeKeyToVolume = async (key: string) => {
+
+    const args1 = ["pull", "vonwig/function_write_files"];
+    const args2 = [
+        "run",
+        "-v",
+        "openai_key:/root",
```

Comment:
Same as above ... I think we might break ourselves if we change the Docker build in such a way that /root is no longer the process $HOME